File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # iSum = 0
        # aFound = []
        # for iii in range(len(nums)):
        #     for jjj in range(len(nums)):
        #         if iii == jjj:
        #             pass
        #         elif (nums[iii] + nums[jjj]) == target:
        #             return [iii, jjj]

        # optimise solution
        bufferDict ={}
        for i in range(len(nums)):
            if nums[i] in bufferDict:
                return [bufferDict[nums[i]],i]
            else:
                bufferDict[target- nums[i]]=i

File: 0001-two-sum/0001-two-sum.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut return_vec: Vec<i32> = Vec::new();
        for index in 0..nums.len(){
            
            for i in index+1..nums.len(){

                let tar :i32 =  nums[index]+nums[i];
                // print!("{:?}: ",nums[index]);
                if(tar == target ){
                    return_vec.push(index.try_into().unwrap());
                    return_vec.push(i.try_into().unwrap());
                    print!(" {:?}: {:?}",index,i);
                    return return_vec;
                }
                
            }
        }

        return return_vec;
    }
}